File: services/reasoning-engine/src/mctagents/core/protocol/evidence.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from .enums import SourceType


class Evidence(BaseModel):
    id: str = Field(..., description="Unique identifier (UUID)")
    run_id: str = Field(..., description="Reference to the parent run (UUID)")
    source_type: SourceType
    source_ref: str | None = Field(
        default=None, description="Reference URI or path to the source"
    )
    summary: str = Field(..., description="Brief summary of the evidence")
    full_text: str | None = Field(default=None, description="Full text content")
    reliability_score: float = Field(default=0.5, ge=0.0, le=1.0)
    source_authority: float | None = Field(default=None, ge=0.0, le=1.0)
    recency: float | None = Field(default=None, ge=0.0, le=1.0)
    specificity: float | None = Field(default=None, ge=0.0, le=1.0)
    independence: float | None = Field(default=None, ge=0.0, le=1.0)
    retrieval_confidence: float | None = Field(default=None, ge=0.0, le=1.0)
    chunk_id: str | None = Field(
        default=None, description="Reference to document chunk (UUID)"
    )
    document_id: str | None = Field(
        default=None, description="Reference to source document (UUID)"
    )
    supports_claim_ids: list[str] = Field(
        default_factory=list, description="Claim IDs this evidence supports"
    )
    attacks_claim_ids: list[str] = Field(
        default_factory=list, description="Claim IDs this evidence attacks"
    )
    metadata: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime = Field(default_factory=datetime.now)

File: packages/core-protocol/src/types/evidence.ts
import type { SourceType } from "./enums.js";

export interface Evidence {
  id: string;
  run_id: string;
  source_type: SourceType;
  source_ref: string | null;
  summary: string;
  full_text: string | null;
  reliability_score: number;
  source_authority: number | null;
  recency: number | null;
  specificity: number | null;
  independence: number | null;
  retrieval_confidence: number | null;
  chunk_id: string | null;
  document_id: string | null;
  supports_claim_ids: string[];
  attacks_claim_ids: string[];
  metadata: Record<string, unknown>;
  created_at: string;
}
